modules/signatures.js

```diff
@@ -84,11 +84,46 @@ Signatures.prototype.onBind = function(scope) {
 	__private.assetTypes[transactionTypes.SIGNATURE].bind(scope.accounts);
 };
 
+__private.processPostResult = function(err, res, cb) {
+	var error = null;
+	var response = null;
+
+	// TODO: Need to improve error handling so that we don't
+	// need to parse the error message to determine the error type.
+	var processingError = /(error|processing)/gi;
+	var badRequestBodyError = /(invalid|signature)/gi;
+
+	if (err) {
+		error = new ApiError(err, apiCodes.PROCESSING_ERROR);
+	} else if (res.success) {
+		response = { status: 'Signature Accepted' };
+	} else if (processingError.exec(res.message).length === 2) {
```

Comment:
`.exec` will return array or `null` on mismatch, so it can result with `Cannot read property 'length' of null`
Example: https://runkit.com/4miners/5a8ab71f0fefc50012459cc5

Reply:
Good point. I think using regular expressions here was a bad idea anyway. I'll try to rework this logic to not rely on regular expressions at all if possible.

Reply:
In this case, the errors are strings in the form `Error processing signature: ${err}` so I kept regular expressions for now; I made them more specific though to avoid potential false positive matches. I think we should raise a separate issue about improving error handling in general though since this affects other parts of the code too.